sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -293,8 +293,7 @@ class SparkSession private(
    *
    * @since 2.0.0
    */
-  def createDataFrame[A <: Product : TypeTag](rdd: RDD[A]): DataFrame = {
-    SparkSession.setActiveSession(this)
+  def createDataFrame[A <: Product : TypeTag](rdd: RDD[A]): DataFrame = withActive {
```

Comment:
I am reasonably sure that there are more methods for which we need to add `withActive`.

Reply:
For example `SparkSession.sql(..)`